src/HelperComponents/PaginationComponent.jsx

```diff
@@ -1,94 +1,81 @@
-import { Component } from "react";
 import { Pagination } from "react-bootstrap";
+import { useState } from "react";
 
-class PaginationComponent extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            activePage: 1,
-        }
-    }
+const PaginationComponent = (props) => {
+    const {onPageChange, lastPage, totalItems, itemsPerPage} = props;
+    const [activePage, setActivePage] = useState(1);
 
-    handlePageChange(page) {
+    const handlePageChange = (page) => {
         console.log('page changed', page)
-        const { onPageChange } = this.props;
-        this.setState({activePage: page});
+        onPageChange;
+        setActivePage(page);
         onPageChange(page);
     }
-
-    render() {
-        return(
-            <div>
-                <Pagination>{this.renderPages()}</Pagination>
-            </div>
-        );
-    }
-
-    renderPages() {
-        const { totalItems, itemsPerPage, lastPage } = this.props;
+    
+    const renderPages = () =>  {
         if (lastPage <= 5) {
-        return this.renderBasicPagination(lastPage);
+        return renderBasicPagination(lastPage);
         }
-        return this.renderAdvancedPagination(lastPage);
+        return renderAdvancedPagination(lastPage);
     }
 
-    renderBasicPagination(lastPage) {
+    const renderBasicPagination = (lastPage) => {
         let pages = [];
         for (let i = 1; i <= lastPage; i++) {
-            pages.push(this.renderPaginationItem(i, i === this.state.activePage));
+            pages.push(renderPaginationItem(i, i === activePage));
         }
         return pages;
     }
-
-    renderAdvancedPagination(lastPage) {
-        const middlePage = this.calculateMidPage(lastPage);
+    const renderAdvancedPagination = (lastPage) => {
+        const middlePage = calculateMidPage(lastPage);
         const preMiddlePage = middlePage - 1;
         const postMiddlePage = middlePage + 1;
-        const { activePage } = this.state;
+        activePage; 
```

Comment:
```suggestion
```

random reference of activePage.. it doesn't do anything here, you already have activePage declared in the function. you can remove this line.